File: src/translate-app/frontend/src/types/session.ts
export interface Session {
  id: string
  title: string
  status: SessionStatus
  targetLang: string
  style: TranslationStyle | ''
  model: string
  createdAt: string
  updatedAt: string
}

export interface Message {
  id: string
  sessionId: string
  role: 'user' | 'assistant'
  displayOrder: number
  displayMode: 'bubble' | 'bilingual' | 'file'
  originalContent: string
  translatedContent: string
  fileId: string | null
  sourceLang: string
  targetLang: string
  style: TranslationStyle
  modelUsed: string
  originalMessageId: string | null
  tokens: number
  fileSize?: number
  createdAt: string
  updatedAt: string
}

export type TranslationStyle = 'casual' | 'business' | 'academic'
export type SessionStatus = 'active' | 'pinned' | 'archived' | 'deleted'

File: src/translate-app/backend/internal/model/session.go
package model

type SessionStatus string

const (
	SessionStatusActive   SessionStatus = "active"
	SessionStatusPinned   SessionStatus = "pinned"
	SessionStatusArchived SessionStatus = "archived"
	SessionStatusDeleted  SessionStatus = "deleted"
)

type Session struct {
	ID         string        `json:"id"`
	Title      string        `json:"title"`
	Status     SessionStatus `json:"status"`
	TargetLang string        `json:"targetLang"`
	Style      string        `json:"style"`
	Model      string        `json:"model"`
	CreatedAt  string        `json:"createdAt"` // RFC3339 (Wails TS binding)
	UpdatedAt  string        `json:"updatedAt"`
}
